scripts/jenkins/Jenkinsfile

```diff
@@ -109,241 +62,292 @@ CMAKE_FLAGS_BUILD_ALL = [
 CMAKE_FLAGS_COVERAGE = ['ENABLE_COVERAGE': 'ON']
 
 CMAKE_FLAGS_CLANG = [
-    'CMAKE_C_COMPILER': 'clang',
-    'CMAKE_CXX_COMPILER': 'clang++'
+    'CMAKE_C_COMPILER': 'clang-5.0',
+    'CMAKE_CXX_COMPILER': 'clang++-5.0'
```

Comment:
What about clang-6?

Reply:
we had a ton of discussions already about what clang version to use. it was decided that we should use 5.0. no idea why you bring it up anew? if you want to add more versions do so.

Reply:
We had? I only remember discussing clang together with ASAN problems.

Reply:
Yes I remember you specifically asking me to move the clang version to 5.0 as a result of that. Hence I ported the old build into the new infrastructure and moved to debian unstable and installed clang 5.0.

Reply:
Ok, I understand. For me the discussion was only about the clang-asan build job, not about the clang in general. I thought clang-6.0 is obvious anyway because afaik we already had jobs running it (clang on unstable).

I asked here specifically, because in the Jenkinsfile "clang" alone should have been enough (if one Docker image installs clang-5 and another clang-6). But you are the expert about which is the best approach to have different build configurations in our setup.

In this PR you should as little changes as possible anyway, the threads of discussions are quite confusing already.

Reply:
> I thought clang-6.0 is obvious anyway because afaik we already had jobs running it (clang on unstable).

I do not know of any such build job. The build server only used clang 3.8.

Reply:
Sorry, my wrong assumption. Then it is definitely a new feature.